src/NuGet.Clients/NuGet.PackageManagement.UI/Utility/UIUtility.cs

```diff
@@ -47,8 +47,8 @@ private static bool IsHttpUrl(Uri uri)
             return (uri.Scheme == Uri.UriSchemeHttp || uri.Scheme == Uri.UriSchemeHttps);
         }
 
-        private static readonly string[] _scalingFactor = new string[] {
-            String.Empty,
+        private static readonly string[] ScalingFactor = new string[] {
```

Comment:
Shouldn't this stay as `_scalingFactor`?
https://github.com/NuGet/NuGet.Client/blob/dev/docs/coding-guidelines.md#c-coding-style

Reply:
This is IDE suggested fix.

Reply:
I think this whole algorithm is too complex and there's a bug here where billions will look like `1.23G` because `G` is specified for billions instead of `B`.  Does this scaling factor need to know about byte size and item count?  If so then it would need to differentiate them.

I think we should simplify this method and get rid of the loop which is confusing and made you add this silly looking if statement:
```
// If number is 999561 then v = 999.56 so with G3 which result in 1000k ~ 1E+03K.
if (v > 999)
{
    v = 999;
}
```

What do you think about this instead?
```C#
public static string NumberToString(long number, IFormatProvider culture)
{
    const double thousand = 1000D;
    const double million = 1000000D;
    const double billion = 1000000000D;
    const double trillion = 1000000000000D;

    if (number >= trillion)
    {
        return (number / trillion).ToString("0.##T", culture);
    }

    if (number >= billion)
    {
        return (number / billion).ToString("0.##B", culture);
    }

    if (number >= million)
    {
        return (number / million).ToString("0.##M", culture);
    }

    if (number >= thousand)
    {
        return (number / thousand).ToString("0.##K", culture);
    }

    return number.ToString("G0", culture);
}
```

Then you can get rid of the `ScalingFactor` array as well and this method makes waaaaay more sense.

We might need to methods:
`ByteCountToString()`: Returns the number as bytes so KB, MB, GB, TB (kilobytes, megabytes, gigabytes, terabytes)
`CountToString()`: Returns a number as a count so K, M, B, T (thousand, million, billion, trillion)

Reply:
> because G is specified for billions instead of B.

G is the SI (standard international) prefix. If we want to use B in English, then we need to make it a string resource to allow the translators to localize this into the correct value for each language.  So that part of it is by-design, I think.  See also https://github.com/NuGet/Home/issues/10864

Reply:
Well yeah while we're here we might as well localize it!  Fix two bugs with one stone

Reply:
@jeffkl @zivkan 
I took Jeff's code as base and added localization text, please check now.
Number - Previous experience - New experience
![image](https://user-images.githubusercontent.com/8766776/132932525-c237e98c-d24f-4c6a-92d8-57750b7f27eb.png)